src/main/java/frc/trigon/robot/subsystems/swerve/Swerve.java

```diff
@@ -165,17 +164,22 @@ public void selfRelativeFeedForwardDrive(ChassisSpeeds chassisSpeeds, DriveFeedf
     }
 
     /**
-     * Locks the swerve, so it'll be hard to move it. This will make the modules look in the middle of a robot in an "x" shape.
+     * Drives the swerve to a certain angular velocity. Used for Wheel Radius Characterization.
+     *
+     * @param omegaRadiansPerSecond the target angular velocity in radians per second
      */
-    void lockSwerve() {
-        final SwerveModuleState
-                right = new SwerveModuleState(0, Rotation2d.fromDegrees(-45)),
-                left = new SwerveModuleState(0, Rotation2d.fromDegrees(45));
+    public void runWheelRadiusCharacterization(double omegaRadiansPerSecond) {
+        selfRelativeDrive(new ChassisSpeeds(0, 0, omegaRadiansPerSecond));
+    }
 
-        swerveModules[0].setTargetState(left);
-        swerveModules[1].setTargetState(right);
-        swerveModules[2].setTargetState(right);
-        swerveModules[3].setTargetState(left);
+    void initializeDrive(boolean shouldUseClosedLoop) {
+        previousSetpoint = new SwerveSetpoint(getSelfRelativeVelocity(), getModuleStates(), DriveFeedforwards.zeros(PathPlannerConstants.getRobotConfig().numModules));
```

Comment:
are the FFs zero?
you're calling PathPlannerConstants.getRobotConfig() too much

Reply:
wdym are the FF's 0?

Reply:
Third param

Reply:
Third param